app/src/main/cpp/skyline/kernel/svc.cpp

```diff
@@ -21,374 +22,466 @@ namespace skyline::kernel::svc {
+                state.ctx->gpr.w0 = result::InvalidCoreId;
+                return;
+            }
+
+            if (affinityMask.none() || !affinityMask.test(coreId)) {
+                state.logger->Warn("svcSetThreadCoreMask: 'affinityMask' invalid: {} (Ideal Core: {})", affinityMask, coreId);
+                state.ctx->gpr.w0 = result::InvalidCombination;
+                return;
+            }
+
+            state.logger->Debug("svcSetThreadCoreMask: Setting thread #{}'s Ideal Core ({}) + Affinity Mask ({})", thread->id, coreId, affinityMask);
+
+            thread->idealCore = coreId;
+            thread->affinityMask = affinityMask;
+
+            state.ctx->gpr.w0 = Result{};
+        } catch (const std::out_of_range &) {
+            state.logger->Warn("svcSetThreadCoreMask: 'handle' invalid: 0x{:X}", handle);
+            state.ctx->gpr.w0 = result::InvalidHandle;
+        }
+    }
+
+    void GetCurrentProcessorNumber(const DeviceState &state) {
+        state.logger->Debug("svcGetCurrentProcessorNumber: Writing current core for thread #{}: {}", state.thread->id, state.thread->coreId);
+        state.ctx->gpr.w0 = state.thread->coreId;
+    }
+
+    void ClearEvent(const DeviceState &state) {
+        auto object{state.process->GetHandle<type::KEvent>(state.ctx->gpr.w0)};
         object->signalled = false;
-        state.ctx->registers.w0 = Result{};
+        state.ctx->gpr.w0 = Result{};
     }
 
-    void MapSharedMemory(DeviceState &state) {
+    void MapSharedMemory(const DeviceState &state) {
         try {
-            auto object{state.process->GetHandle<type::KSharedMemory>(state.ctx->registers.w0)};
-            auto address{state.ctx->registers.x1};
+            auto object{state.process->GetHandle<type::KSharedMemory>(state.ctx->gpr.w0)};
+            auto pointer{reinterpret_cast<u8 *>(state.ctx->gpr.x1)};
 
-            if (!util::PageAligned(address)) {
-                state.ctx->registers.w0 = result::InvalidAddress;
-                state.logger->Warn("svcMapSharedMemory: 'address' not page aligned: 0x{:X}", address);
+            if (!util::PageAligned(pointer)) {
+                state.ctx->gpr.w0 = result::InvalidAddress;
+                state.logger->Warn("svcMapSharedMemory: 'pointer' not page aligned: 0x{:X}", pointer);
                 return;
             }
 
-            auto size{state.ctx->registers.x2};
+            size_t size{state.ctx->gpr.x2};
             if (!util::PageAligned(size)) {
-                state.ctx->registers.w0 = result::InvalidSize;
+                state.ctx->gpr.w0 = result::InvalidSize;
                 state.logger->Warn("svcMapSharedMemory: 'size' {}: 0x{:X}", size ? "not page aligned" : "is zero", size);
                 return;
             }
 
-            auto permission{*reinterpret_cast<memory::Permission *>(&state.ctx->registers.w3)};
+            auto permission{*reinterpret_cast<memory::Permission *>(&state.ctx->gpr.w3)};
```

Comment:
The reinterpret cast can be dropped here

Reply:
I found an unrelated issue due to this, I must've assumed that the rwx fields were bitfield booleans (I remember being them like this), it must've changed at some point and broken them. I've now fixed it by making `Permission` a `union` and having an appropriate constructor.